tests/test_integrations.py

```diff
@@ -992,8 +992,17 @@ def _echo(args):
 @pytest.mark.parametrize(
     "cmd, exp_rtn",
     [
+        ("2+2", 0),
         ("import sys; sys.exit(0)", 0),
         ("import sys; sys.exit(100)", 100),
+        ("@('exit')", 0),
+        ("exit 100", 100),
+        ("exit()", 0),
+        ("exit(100)", 100),
+        ("__xonsh__.exit=0", 0),
+        ("__xonsh__.exit=100", 100),
+        ("raise Exception()", 1),
+        ("raise SystemExit(100)", 100),
```

Comment:
Nice coverage. What about `exit 'foo'` or `exit('foo')`? I know bash doesn't support it:

```
 ~ 🐚 bash -c 'exit foo'
bash: line 1: exit: foo: numeric argument required
```

Is it worthwhile to capture that expectation also, at least to make sure it doesn't break assumptions about the type handling?